microsoft-edge/webview2/concepts/threading-model.md

```diff
@@ -32,6 +32,52 @@ The only exception is for the `Content` property of `CoreWebView2WebResourceRequ
 
 Callbacks including event handlers and completion handlers run serially.  
 After you run an event handler and begin a message loop, you're unable to run any event handler or completion callback in a re-entrant manner.  
+If you use methods that will spin up nested message loops such as `ShowDialog` synchronously within a WebView event handler, it will lead to reentrancy and leave the event handler in stack indefinitely.
+
+```csharp
+private void Btn_Click(object sender, EventArgs e)
+{
+   // Post web message when button is clicked
+   this.webView2Control.ExecuteScriptAsync("window.chrome.webview.postMessage(\"Open Dialog\");");
+}
+
+private void CoreWebView2_WebMessageReceived(object sender, CoreWebView2WebMessageReceivedEventArgs e)
+{
+   string msg = e.TryGetWebMessageAsString();
+   if (msg == "Open Dialog")
+   {
+      Form1 form = new Form1(); // Create a new form that contains a new WebView when web message is received.
+      form.ShowDialog(); // This will cause reentrancy issue and cause the newly created WebView inside the modal dialog to hang.
+   }
+}
+```  
+      > [!NOTE]
```

Comment:
Is this note supposed to be in the middle of your examples?

Reply:
@jasonstephen15 - who is the above question directed to?

Reply:
@tofuandeve ^

Reply:
You probably want to left-indent the Note block.  That will fix the Warning about "bad code tagging".
You probably want to move the Note to above or below the two code listings.